src/PrestaShopBundle/ApiPlatform/Metadata/CQRSPartialUpdate.php

```diff
@@ -0,0 +1,131 @@
+class CQRSPartialUpdate extends CQRSCommand
+{
+    public function __construct(
+        ?string $uriTemplate = null,
+        ?array $types = null,
+        $formats = null,
+        $inputFormats = null,
+        $outputFormats = null,
+        $uriVariables = null,
+        ?string $routePrefix = null,
+        ?string $routeName = null,
+        ?array $defaults = null,
+        ?array $requirements = null,
+        ?array $options = null,
+        ?bool $stateless = null,
+        ?string $sunset = null,
+        ?string $acceptPatch = null,
+        $status = null,
+        ?string $host = null,
+        ?array $schemes = null,
+        ?string $condition = null,
+        ?string $controller = null,
+        ?array $cacheHeaders = null,
+
+        ?array $hydraContext = null,
+        ?array $openapiContext = null,
+        ?bool $openapi = null,
+        ?array $exceptionToStatus = null,
+
+        ?bool $queryParameterValidationEnabled = null,
+
+        ?string $shortName = null,
+        ?string $class = null,
+        ?bool $paginationEnabled = null,
+        ?string $paginationType = null,
+        ?int $paginationItemsPerPage = null,
+        ?int $paginationMaximumItemsPerPage = null,
+        ?bool $paginationPartial = null,
+        ?bool $paginationClientEnabled = null,
+        ?bool $paginationClientItemsPerPage = null,
+        ?bool $paginationClientPartial = null,
+        ?bool $paginationFetchJoinCollection = null,
+        ?bool $paginationUseOutputWalkers = null,
+        ?array $paginationViaCursor = null,
+        ?array $order = null,
+        ?string $description = null,
+        ?array $normalizationContext = null,
+        ?array $denormalizationContext = null,
+        ?string $security = null,
+        ?string $securityMessage = null,
+        ?string $securityPostDenormalize = null,
+        ?string $securityPostDenormalizeMessage = null,
+        ?string $securityPostValidation = null,
+        ?string $securityPostValidationMessage = null,
+        ?string $deprecationReason = null,
+        ?array $filters = null,
+        ?array $validationContext = null,
+        $input = null,
+        $output = null,
+        $mercure = null,
+        $messenger = null,
+        ?bool $elasticsearch = null,
+        ?int $urlGenerationStrategy = null,
+        ?bool $read = null,
+        ?bool $deserialize = null,
+        ?bool $validate = null,
+        ?bool $write = null,
+        ?bool $serialize = null,
+        ?bool $fetchPartial = null,
+        ?bool $forceEager = null,
+        ?int $priority = null,
+        ?string $name = null,
+        $provider = null,
+        $processor = null,
+        array $extraProperties = [],
+        ?string $CQRSCommand = null,
+        ?string $CQRSQuery = null,
+        array $scopes = [],
+        ?array $CQRSQueryMapping = null,
+        ?array $ApiResourceMapping = null,
+        ?array $CQRSCommandMapping = null,
+    ) {
```

Comment:
I recently wrote an article about thi https://github.com/PululuK/php_function_infinite_variables 

I recommend that the function receives an array, for example a [ParameterBag](https://github.com/symfony/symfony/blob/7.1/src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBagInterface.php) and manage parameter validation with an [optionsResolver](https://symfony.com/doc/current/components/options_resolver.html) for example

Reply:
Yep in another context I would have done something similar, however, in this case, this operation is completely based on how ApiPlatform handles its own operations see their default class https://github.com/api-platform/core/blob/2.7/src/Metadata/HttpOperation.php

Overriding this internal class was quite a challenge, and for our custom operations to work correctly and be interpreted the right way by the framework we had to follow the same conventions: multiple very long list of parameters, as little change in parameters order as possible, keeping the convention of getters/setters

This ay ApiPlatform is able to build our operation classes the same way it's building its own

Reply:
The option resolver could be interesting, but most of the validation is already handed thanks to the strongly typed arguments anyway, option resolvers are indeed interesting when passing an array of options as it's mostly scalar non-typed data But here since we follow the ApiPlatform way at least we have the benefit of the strong typing